dash/include/dash/Pair.h

```diff
@@ -195,16 +195,37 @@ namespace dash {
     x.swap(y);
   }
 
+  /**
+   * Internal helper to strip types
+   *
+   * Duplicated here to not rely on an STL implementation
+   */
+  template <class T>
+  struct __make_pair_return_impl
+  {
+    typedef T type;
+  };
+  template <class T>
+  struct __make_pair_return_impl<std::reference_wrapper<T>>
+  {
+      typedef T& type;
+  };
+  template <class T>
+  struct __make_pair_return
+  {
+      typedef typename __make_pair_return_impl<typename std::decay<T>::type>::type type;
+  };
+
```

Comment:
Instead of the preceding underscores, please put `make_pair_return` (without underscores) in namespace `dash::internal`. Also, the naming is not entirely intuitive. Why not call it `dash::internal::decay_and_strip`? (so it's clear that this is our version of the GNU `std::__decay_and_strip`)